extensions/wikia/PhalanxII/maintenance/MigratePhalanxUserBlocks.php

```diff
@@ -0,0 +1,184 @@
+ */
+class MigratePhalanxUserBlocks extends Maintenance {
+	private $batchSize = 1000;
+	private $curBlockId = 0;
+	private $dryRun = false;
+
+	public function __construct() {
+		parent::__construct();
+		$this->addOption( 'dry-run', "Perform a dry run, don't close any accounts" );
+	}
+
+	public function execute() {
+		$this->dryRun = $this->getOption( 'dry-run', false );
+
+		$this->setCurrentUser();
+
+		do {
+			$userBlocks = $this->getUserBlocks();
+
+			if ( empty( $userBlocks ) ) {
+				break;
+			}
+
+			$blockIdsClosed = $this->closeAccounts( $userBlocks );
+
+			$this->deletePhalanxBlocks( $blockIdsClosed );
+
+			$this->output( count( $blockIdsClosed ) . " user accounts closed.\n" );
+		} while ( !empty( $userBlocks ) );
+
+		$this->reloadPhalanxData();
+	}
+
+	private function getUserBlocks() {
+		$dbr = $this->getPhalanxDB();
+
+		$userBlocks = [];
+
+		$res = $dbr->select(
+			'phalanx',
+			[ 'p_id', 'p_text' ],
+			[
+				'p_type' => 8,
+				'p_exact' => 1,
+				'p_regex' => 0,
+				'p_expire IS NULL',
+				'p_ip_hex IS NULL',
+				'p_timestamp < ' . $dbr->addQuotes( date( 'YmdHis', strtotime( '-3 months' ) ) ),
+				'p_id > ' . $dbr->addQuotes( $this->curBlockId ),
+			],
+			__METHOD__,
+			[
+				'LIMIT' => $this->batchSize,
+			]
+		);
+
+		$blockId = 0;
+		foreach ( $res as $row ) {
+			$blockId = $row->p_id;
+			if ( !empty( $row->p_text ) && !IP::isIPAddress( $row->p_text ) ) {
+				$userBlocks[$blockId] = $row->p_text;
+			}
+		}
+
+		$this->curBlockId = $blockId;
+
+		return $userBlocks;
+	}
+
+	private function closeAccounts( $userBlocks ) {
+		$blockIdsClosed = [];
+
+		if ( empty( $userBlocks ) ) {
+			return $blockIdsClosed;
+		}
+
+		foreach ( $userBlocks as $blockId => $userName ) {
+			$userObj = User::newFromName( $userName );
+
+			if ( $userObj instanceof User && $userObj->getId() != 0 ) {
+				if ( !$this->isClosed( $userObj ) ) {
+					$success = $this->closeUserAccount( $userObj );
+					if ( !$success ) {
+						continue;
+					}
+				}
+
+				$blockIdsClosed[] = $blockId;
+			} else {
+				$this->output( "User block {$userName} for ID {$blockId} does not match an existing user!\n" );
+			}
+		}
+
+		return $blockIdsClosed;
+	}
+
+	private function closeUserAccount( User $userObj ) {
+		$this->output( "Closing account {$userObj->getName()} ... " );
+
+		$closeReason = 'Closing perma Phalanx blocked accounts';
+		$statusMsgOne = '';
+		$statusMsgTwo = '';
+		$keepEmail = true;
+
+		if ( !$this->dryRun ) {
+			$success = EditAccount::closeAccount( $userObj, $closeReason, $statusMsgOne, $statusMsgTwo, $keepEmail );
+		} else {
+			$this->output( "skipped (dry run)\n" );
+			return true;
+		}
+
+		if ( !$success ) {
+			$this->output( "failed\n" );
+			return false;
+		}
+
+		$this->output( "success\n" );
+		return true;
+	}
+
+	private function deletePhalanxBlocks( $blockIds ) {
+		if ( $this->dryRun ) {
+			return true;
+		}
+
+		$dbw = $this->getPhalanxDB( DB_MASTER );
+
+		$res = $dbw->delete(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$res". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)